ui.apps/src/main/content/jcr_root/apps/etoolbox-rollout-manager/clientlibs/rollout-manager-ui/js/rollout-utils.js

```diff
@@ -60,38 +78,77 @@
         }
     }
 
-    async function createStatusUpdater(logger, taskId) {
-        let response = { status: 'active' };
-        let offset = 0;
-        while (response.status === 'active') {
-            response = await getStatusInfo(taskId, offset);
-
-            if (response.error) throw new Error(`${response.error}`);
-
-            if (response.messages && response.messages.length) {
-                offset = response.messages.reduce((total, msg) => {
-                    logger.log(msg);
-                    return msg.id > total ? msg.id : total;
-                }, offset);
-            }
-
-            if (!response.status) throw new Error('Wrong response from the server');
+    async function createStatusUpdater(logger, startIdArray = [], offsetFromStart = false) {
+        let response = { tasks: [{'status': 'active'}]};
+        while (response.tasks && response.tasks.some(item => item.status === 'active')) {
+            const data = ns.getItemsData();
+            if (!data.length) return;
 
+            const id = data.map(item => item.id).join(';');
+            const offset =  data.map(item => {
+                if (!offsetFromStart) return item.offset;
+                return isOpenDialogTask(item.id) ? 0 : item.offset
+            }).join(';');
+            const currentIdArray = id.split(';');
+            const isAbortRequest = currentIdArray.some(item => !startIdArray.includes(item));
```

Comment:
If `startIdArray` can potentially be large, then from a performance view, it is better to do it this way
```suggestion
            const startIdSet = new Set(startIdArray);
            const currentIdArray = id.split(';');
            const isAbortRequest = currentIdArray.some((item) => !startIdSet.has(item))
```

Reply:
makes sense)
thank you!